dlib/any/any.h

```diff
@@ -28,146 +25,74 @@ namespace dlib
 
     class any
     {
-
     public:
-
-        any()
+        any()                               = default;
+        any(const any& other)               = default;
+        any& operator=(const any& other)    = default;
+        any(any&& other)                    = default;
+        any& operator=(any&& other)         = default;
+
+        template<
+            typename T,
+            typename T_ = std::decay_t<T>,
+            std::enable_if_t<!std::is_same<T_, any>::value, bool> = true
```

Comment:
```suggestion
            std::enable_if_t<!std::is_same<std::decay_t<T>, any>::value, bool> = true
```
That doesn't work?

Reply:
Yes of course, i think that was a bit of copy paste

Reply:
Though I'm sure MSVC will at some point complain about completely valid C++

Reply:
Ha, yeah maybe.  Get the other instances of this too in there.  Looked like there were several.